src/webgpu/api/operation/vertex_state/index_format.spec.ts

```diff
@@ -0,0 +1,380 @@
+export const description = `
+Test indexing, index format and primitive restart.
+`;
+
+import { params, poptions } from '../../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+import { getTextureCopyLayout } from '../../../util/texture/layout.js';
+
+const kHeight = 4;
+const kWidth = 8;
+const kTextureFormat = 'r8uint' as const;
```

Comment:
small change: I changed this to r8uint instead of bgra8unorm because it simplifies the code slightly.